### PR TITLE
docs: Update Link docs

### PR DESCRIPTION
**Pull request by rodrigoeidelvein**
## Motivations

Updating Link docs, it's web only.

### Added

- 

### Changed

- Updating the docs for Link

### Deprecated

- 

### Removed

- 

### Fixed

- 

### Security

- 

## Testing

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

Random photo of Atlantis

### CODE REVIEW

File: docs/components/Link/Link.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+Link should not be used to submit form data, trigger actions, or alter the state
+of the page. Use a [Button](../?path=/docs/components-actions-button-docs--page)
+for these purposes.
+
+## Content guidelines
+
+Contents of the link should be textual, and may include some formatting features
+(e.g. italic text for emphasis may be appropriate, changing the inherited
+underline probably isn't). The text can be plain, from a `Typography` component,
```
Comment (dlrussel) [inline docs/components/Link/Link.stories.mdx:24]: I think we should make this reference to the `typography component a link to that section of the documentation to follow the above pattern like we did with button.

File: docs/components/Link/Web.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    viewMode: "story",
+    previewTabs: { code: { hidden: false } },
+  },
+} as ComponentMeta<typeof Link>;
+
+const BasicTemplate: ComponentStory<typeof Link> = args => (
+  <Link {...args}>What is a Link anyway?</Link>
+);
+
+export const Horizontal = BasicTemplate.bind({});
```
Comment (dlrussel) [inline docs/components/Link/Web.stories.tsx:18]: Horizontal seems like an odd name for this. There isn't a vertical variation and I'm not sure it's descriptive of this example as it just seems to be a basic link. Should we rename it?

**Review by dlrussel — CHANGES_REQUESTED** [review]

File: docs/components/Link/Web.stories.tsx (hunk 2 above)
Comment (rodrigoeidelvein) [reply docs/components/Link/Web.stories.tsx:18]: lol I copied it from `Divider` and forgot to change this const name, thanks for catching it

File: docs/components/Link/Link.stories.mdx (hunk 1 above)
Comment (rodrigoeidelvein) [reply docs/components/Link/Link.stories.mdx:24]: I agree, good idea!

File: docs/components/Link/Web.stories.tsx (hunk 2 above)
Comment (dlrussel) [reply docs/components/Link/Web.stories.tsx]: ah ok sweet!

**Review by dlrussel — APPROVED** [review]
Looks good 🥳

**Pull request merged**